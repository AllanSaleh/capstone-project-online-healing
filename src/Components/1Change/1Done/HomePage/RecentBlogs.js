import React from 'react';
import Tech from './Tech.svg';
import Trauma from './Trauma.svg';
import Prev from './Prev.svg';
import Next from './Next.svg';

export default function RecentBlogs() {
  const blogs=[
    { title: 'Mental Health & Technology', img: Tech },
    { title: 'Healing Trauma', img: Trauma },
    { title: 'Mental Health & Technology', img: Tech },
    { title: 'Healing Trauma', img: Trauma },
    { title: 'Mental Health & Technology', img: Tech },
    { title: 'Healing Trauma', img: Trauma },
  ]

  return (
    <div className="flex flex-col justify-evenly h-section px-sides bg-blue-light">
      <div className="text-title">RECENT BLOGS</div>

      <div className="flex justify-between items-center">
        <button
          onClick={() => {
            const NewPos = document.getElementById('Menu').scrollLeft - 560;
            document.getElementById('Menu').scrollTo({
              left: NewPos,
              behavior: 'smooth',
            });
          }}
          type="button"
          className="mr-4 h-12 w-12 px-3 text-subtitle bg-blue-light rounded-lg border-2 border-black hover:bg-blue-dark"
        >
          <img src={Prev} alt="Previous" />
        </button>

        <div className="relative h-80 w-full overflow-hidden">
          <div
            id="Menu"
            className="h-96 box-border whitespace-nowrap overflow-x-auto overflow-y-hidden"
          >
            {blogs.map((blog, i) => (
              <div
                onClick={(e) =>
                  alert(`Open blog number ${e.target.closest('.relative').id}`)
                }
                id={i}
                className="relative inline-block w-2/6 h-80 box-border mx-12"
              >
                <img
                  className="rounded-lg absolute w-full cursor-pointer"
                  src={blog.img}
                  alt="Background"
                />

                <div className="w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-8 left-4 text-white text-paragraph bg-black bg-opacity-50">
                  {blog.title}
                </div>
              </div>
            ))}
          </div>
        </div>

        <button
          onClick={() => {
            const NewPos = document.getElementById('Menu').scrollLeft + 560;
            document.getElementById('Menu').scrollTo({
              left: NewPos,
              behavior: 'smooth',
            });
          }}
          type="button"
          className="ml-4 h-12 w-12 px-3 text-subtitle bg-blue-light rounded-lg border-2 border-black hover:bg-blue-dark"
        >
          <img src={Next} alt="Previous" />
        </button>
      </div>
    </div>
  );
}
