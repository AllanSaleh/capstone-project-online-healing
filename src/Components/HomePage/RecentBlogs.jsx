import React from 'react';

import Tech from './Images/Tech.svg';
import Trauma from './Images/Trauma.svg';
import Prev from './Images/Prev.svg';
import Next from './Images/Next.svg';

export default function RecentBlogs() {
  const blogs = [
    { title: 'Mental Health & Technology', img: Tech },
    { title: 'Healing Trauma', img: Trauma },
    { title: 'Mental Health & Technology', img: Tech },
    { title: 'Healing Trauma', img: Trauma },
    { title: 'Mental Health & Technology', img: Tech },
    { title: 'Healing Trauma', img: Trauma },
  ];

  return (
    <div className="flex flex-col justify-evenly h-64 lg:h-section px-sides bg-blue-light">
      <div className="text-3xl lg:text-title">RECENT BLOGS</div>

      <div className="flex justify-between items-center">
        <button
          onClick={() => {
            const NewPos = document.getElementById('Menu').scrollLeft - 500;
            document.getElementById('Menu').scrollTo({
              left: NewPos,
              behavior: 'smooth',
            });
          }}
          type="button"
          className="mr-2 lg:mr-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
        >
          <img src={Prev} alt="Previous" />
        </button>

        <div className="relative h-40 lg:h-80 w-full overflow-hidden">
          <div
            id="Menu"
            className="h-44 lg:h-96 box-border whitespace-nowrap overflow-x-auto overflow-y-hidden"
          >
            {blogs.map((blog, i) => (
              <div
                onClick={(e) => alert(`Open blog number ${e.target.closest('.relative').id}`)}
                id={i}
                className="relative inline-block w-11/12 lg:w-2/6 h-40 lg:h-80 box-border mx-2 lg:mx-12"
              >
                <img
                  className="rounded-lg absolute h-full w-full cursor-pointer"
                  src={blog.img}
                  alt="Background"
                />

                <div className="w-24 lg:w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-6 left-2 lg:bottom-8 lg:left-4 text-white text-xs lg:text-paragraph bg-black bg-opacity-50">
                  {blog.title}
                </div>
              </div>
            ))}
          </div>
        </div>

        <button
          onClick={() => {
            const NewPos = document.getElementById('Menu').scrollLeft + 500;
            document.getElementById('Menu').scrollTo({
              left: NewPos,
              behavior: 'smooth',
            });
          }}
          type="button"
          className="ml-2 lg:ml-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
        >
          <img src={Next} alt="Previous" />
        </button>
      </div>
    </div>
  );
}
