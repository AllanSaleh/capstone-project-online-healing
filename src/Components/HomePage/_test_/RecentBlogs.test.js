import * as React from 'react';
import RecentBlogs from "../RecentBlogs";
import * as renderer from "react-test-renderer";

it('Recent Blog renders correctly', () => {
  const tree = renderer
    .create(<RecentBlogs/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});