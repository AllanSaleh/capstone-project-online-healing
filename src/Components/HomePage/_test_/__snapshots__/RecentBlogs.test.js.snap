// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent Blog renders correctly 1`] = `
<div
  className="flex flex-col justify-evenly h-64 lg:h-section px-sides bg-blue-light"
>
  <div
    className="text-2xl lg:text-title"
  >
    RECENT BLOGS
  </div>
  <div
    className="flex justify-between items-center"
  >
    <button
      className="mr-2 lg:mr-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Previous"
        src="Prev.svg"
      />
    </button>
    <div
      className="relative h-40 lg:h-80 w-full overflow-hidden"
    >
      <div
        className="h-44 lg:h-96 box-border whitespace-nowrap overflow-x-auto overflow-y-hidden"
        id="Menu"
      >
        <div
          className="relative inline-block w-11/12 lg:w-2/6 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id="Mental Health & Technology"
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="rounded-lg absolute h-full w-full cursor-pointer"
            src="Tech.svg"
          />
          <div
            className="w-24 lg:w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-6 left-2 lg:bottom-8 lg:left-4 text-white text-xs lg:text-paragraph bg-black bg-opacity-50"
          >
            Mental Health & Technology
          </div>
        </div>
        <div
          className="relative inline-block w-11/12 lg:w-2/6 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id="Healing Trauma"
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="rounded-lg absolute h-full w-full cursor-pointer"
            src="Trauma.svg"
          />
          <div
            className="w-24 lg:w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-6 left-2 lg:bottom-8 lg:left-4 text-white text-xs lg:text-paragraph bg-black bg-opacity-50"
          >
            Healing Trauma
          </div>
        </div>
        <div
          className="relative inline-block w-11/12 lg:w-2/6 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id="Mental Health & Technology"
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="rounded-lg absolute h-full w-full cursor-pointer"
            src="Tech.svg"
          />
          <div
            className="w-24 lg:w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-6 left-2 lg:bottom-8 lg:left-4 text-white text-xs lg:text-paragraph bg-black bg-opacity-50"
          >
            Mental Health & Technology
          </div>
        </div>
        <div
          className="relative inline-block w-11/12 lg:w-2/6 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id="Healing Trauma"
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="rounded-lg absolute h-full w-full cursor-pointer"
            src="Trauma.svg"
          />
          <div
            className="w-24 lg:w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-6 left-2 lg:bottom-8 lg:left-4 text-white text-xs lg:text-paragraph bg-black bg-opacity-50"
          >
            Healing Trauma
          </div>
        </div>
        <div
          className="relative inline-block w-11/12 lg:w-2/6 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id="Mental Health & Technology"
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="rounded-lg absolute h-full w-full cursor-pointer"
            src="Tech.svg"
          />
          <div
            className="w-24 lg:w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-6 left-2 lg:bottom-8 lg:left-4 text-white text-xs lg:text-paragraph bg-black bg-opacity-50"
          >
            Mental Health & Technology
          </div>
        </div>
        <div
          className="relative inline-block w-11/12 lg:w-2/6 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id="Healing Trauma"
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="rounded-lg absolute h-full w-full cursor-pointer"
            src="Trauma.svg"
          />
          <div
            className="w-24 lg:w-44 p-2 rounded-lg text-center whitespace-normal cursor-pointer absolute bottom-6 left-2 lg:bottom-8 lg:left-4 text-white text-xs lg:text-paragraph bg-black bg-opacity-50"
          >
            Healing Trauma
          </div>
        </div>
      </div>
    </div>
    <button
      className="ml-2 lg:ml-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Previous"
        src="Next.svg"
      />
    </button>
  </div>
</div>
`;
